Tests/GRPCInterceptorsTests/TracingInterceptorTests.swift

```diff
@@ -66,30 +69,123 @@ final class TracingInterceptorTests: XCTestCase {
         )
       }
 
-      var streamIterator = stream.makeAsyncIterator()
-      var element = await streamIterator.next()
-      XCTAssertEqual(element, "request1")
-      element = await streamIterator.next()
-      XCTAssertEqual(element, "request2")
-      element = await streamIterator.next()
-      XCTAssertNil(element)
-
-      var messages = response.messages.makeAsyncIterator()
-      var message = try await messages.next()
-      XCTAssertEqual(message, ["response"])
-      message = try await messages.next()
-      XCTAssertNil(message)
-
-      let tracer = InstrumentationSystem.tracer as! TestTracer
-      XCTAssertEqual(
-        tracer.getEventsForTestSpan(ofOperationName: methodDescriptor.fullyQualifiedMethod).map {
-          $0.name
-        },
-        [
-          "Request started",
-          "Received response end",
-        ]
+      await AssertStreamContentsEqual(["request1", "request2"], stream)
+      try await AssertStreamContentsEqual([["response"]], response.messages)
+
+      AssertTestSpanComponents(forMethod: methodDescriptor) { events in
+        XCTAssertEqual(
+          events.map({ $0.name }),
+          [
+            "Request started",
+            "Request ended",
+            "Received response start",
+            "Received response end",
+          ]
+        )
+      } assertAttributes: { attributes in
+        XCTAssertEqual(
+          attributes,
+          [
+            "rpc.system": .string("grpc"),
+            "rpc.method": .string(methodDescriptor.method),
+            "rpc.service": .string(methodDescriptor.service.fullyQualifiedService),
+            "rpc.grpc.status_code": .int(0),
+            "server.address": .string("someserver.com"),
+            "server.port": .int(567),
+            "network.peer.address": .string("10.1.2.80"),
+            "network.peer.port": .int(567),
+            "network.transport": .string("tcp"),
+            "network.type": .string("ipv4"),
+          ]
+        )
+      } assertStatus: { status in
+        XCTAssertNil(status)
+      } assertErrors: { errors in
+        XCTAssertEqual(errors, [])
+      }
+    }
+  }
+
+  func testClientInterceptor_UDS() async throws {
```

Comment:
This is an awful lot of duplication. Can we parameterise these tests by the IPv4/IPv6/UDS? If we just pass in the type of address we can let the test set the input and expected values so that the parameters stay easy to read:

```swift
func theTest(_ addressType: AddressType) async throws {
  let localPeer: String
  let remotePeer: String 

  let expectedNetworkType: String
  let ....

  switch addressType {
  case .ipv4:
    // ...
    // etc.
  }

  // ...
}
```